packages/web/components/DataTable/index.tsx

```diff
@@ -259,16 +263,19 @@ export default function DataTable({
         customBodyRender: actions ? actionsColumnRender(actions) : customColumnRender,
       },
     }));
-    setPreparedColumns(pColumns);
-  }, [actionsColumnRender, columns]); // eslint-disable-line
 
+    setPreparedColumns(pColumns);
+  }, [actionsColumnRender, columns, localStorage.getItem(`${storageKey}-column-view`)]); // eslint-disable-line
 
   const onRowsDelete = useCallback((rowsDeleted: any) => {
     confirm({ description: confirmDeleteMessage })
       .then(() => {
         rowsDeleted.data.map(({ dataIndex }: { dataIndex: number }) => doDelete && doDelete(data[dataIndex]));
       })
       .catch();
+
+    localStorage.setItem(`${storageKey}-filters`, JSON.stringify({}));
+    setFilters({});
```

Comment:
why filters are being cleared on row deletion?

Reply:
The original bug is from MuiDataTable itself and this was the only way to fix it here on our side. If the user filtered by a situation (active for example) selected one or more records from the grid and deleted them, the grid would bug (it kept the selected records, but without listing the deleted ones because it was filtering by assets). Applying this change to clear filters when deleting a row resolved the issue.